lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -80,19 +82,30 @@ bool stringBeginsWithDigit(const std::string& inputString)
     return false;
 }
 
-// find position index for a given layer in the local LayerStack ( strong-to-weak order )
-uint32_t findLayerIndex(const UsdStageWeakPtr& stage, const PXR_NS::SdfLayerHandle& layer)
+// This function calculates the position index for a given layer across all
+// the site's local LayerStack.
+uint32_t findLayerIndex(const UsdPrim& prim, const PXR_NS::SdfLayerHandle& layer)
 {
-    const auto& layerStack = stage->GetLayerStack();
-
     uint32_t position { 0 };
 
-    auto iter = std::find(std::begin(layerStack), std::end(layerStack), layer);
+    const PXR_NS::PcpPrimIndex& primIndex = prim.GetPrimIndex();
+
+    // iterate through the expanded primIndex
+    for (const PcpNodeRef& node : primIndex.GetNodeRange()) {
```

Comment:
PF results back and I am hit with `-Wrange-loop-analysis` with Clang on MacOS:

```
    /local/S/jenkins/workspace/ecg-mayausd-branch-preflight-2022-python3-osx/ecg-maya-usd/maya-usd/lib/mayaUsd/ufe/Utils.cpp:94:28: error: loop variable 'node' is always a copy because the range of type 'pxrInternal_v0_20__pxrReserved__::PcpNodeRange' (aka 'pair<pxrInternal_v0_20__pxrReserved__::PcpNodeIterator, pxrInternal_v0_20__pxrReserved__::PcpNodeIterator>') does not return a reference [-Werror,-Wrange-loop-analysis]
        for (const PcpNodeRef& node : primIndex.GetNodeRange()) {
```

I believe I have previously commented on this flag before in https://github.com/Autodesk/maya-usd/pull/930/ 

My original intention here is that I don't want a copy in my range loop and find it hard this iterator object is always copy. 

`for(const PcpNodeRef &node: primIndex.GetNodeRange()) `

Not to mention that this is pattern that is used everywhere in USD code base:

e.g https://github.com/PixarAnimationStudios/USD/blob/dc710925675f7f58f7a37f6c18d046a687b09271/pxr/usd/usdShade/material.cpp#L439

Reply:
Did you try const auto& node?